### ISSUE TITLE
Transformer: Loss goes down but bleu stays at 0.0002056644

### ISSUE THREAD

**Issue by andrewor14** [tensorflow/models#9830]
# Prerequisites

## 1. The entire URL of the file you are using

`https://github.com/tensorflow/models/tree/cdc4cad/official/nlp/transformer/transformer_main.py`

## 2. Describe the bug

Loss goes down steadily but bleu scores (both cased and uncased) stay at `0.00020566437797242543`.

Loss:
```
3.869929790496826
2.5293941497802734
2.2222673892974854
2.081308126449585
1.9934587478637695
1.944015622138977
1.894033432006836
1.8554004430770874
... (still running)
```

## 3. Steps to reproduce

```
python3 transformer_main.py
  --num_gpus=2 --param_set=big --data_dir="$DATA_DIR"
  --model_dir="$MODEL_DIR" --vocab_file="$DATA_DIR/vocab.ende.32768"
  --bleu_source="$DATA_DIR/newstest2014.en" --bleu_ref="$DATA_DIR/newstest2014.de"
  --train_steps=100000 --steps_between_evals=10000 --max_length=64 --batch_size=4096
```

## 4. Expected behavior

Loss should go down, bleu scores should go up and approach 27.

## 5. Additional context

I changed `vocab_size` to 33945 in my `model_params.py` to accurately reflect my vocab file.

Just tried to follow instructions here: `https://github.com/tensorflow/models/blob/cdc4cad/official/nlp/transformer/README.md`

## 6. System information

- OS Platform and Distribution: Ubuntu 18.04
- TensorFlow installed from (source or binary): source
- TensorFlow version (use command below): `1.15.0-rc1-33260-g311e92b1895 2.4.0-rc3`
- Python version: 3.6.9
- Bazel version (if compiling from source): 3.1.0
- GCC/Compiler version (if compiling from source): 7.5.0
- CUDA/cuDNN version: CUDA 10.1, cuDNN 7.6.2
- GPU model and memory: RTX 2080 Ti, 11GB

**Comment by saberkun** [comment]
The TF1 version is not well tested. Sorry, we are not maintaining the code in the git history.
@frederick0329 has updated and well tested new transformer implementation and just added the experiment config.
https://github.com/tensorflow/models/commit/a283424ad02ce174b00bf6345f5bbaf6735722de

**Comment by frederick0329** [comment]
You can run the tf2 code by 
```
pip install tf-models-nightly
```
(will be in 2.5 release)

with the following command
```
python3 models/official/nlp/train.py --experiment=wmt_transformer/larg
e --params_override=runtime.num_gpus=4,runtime.distribution_strategy=mirrored,runtime.mixed_precision_dtype=float16
,runtime.loss_scale=dynamic,task.sentencepiece_model_path='SENTENCE_PEICE_MODEL_PATH' --mode=t
rain_and_eval --model_dir='PATH_TO_SAVE_MODEL'
```
To generate SENTENCE_PEICE_MODEL_PATH, please refer to https://github.com/tensorflow/models/blob/master/official/nlp/data/train_sentencepiece.py .

With 300k steps, you should get around 27.5 BLEU.

**Comment by andrewor14** [comment]
@frederick0329 I see. Thank you for the alternative suggestion.

In case others are running into the same problem with this code, I was able to get a significant bleu score after switching to newstest2015 ([here](https://nlp.stanford.edu/projects/nmt/data/wmt14.en-de/newstest2015.en) and [here](https://nlp.stanford.edu/projects/nmt/data/wmt14.en-de/newstest2015.de)). However, I did run into the overfitting problem [others have mentioned](https://github.com/tensorflow/models/issues/7816), even at large batch sizes such as 24576 and 32768. I will try the new code and circle back.
